### PR TITLE
WIP,BUG: Do not doubly-unquote executable names

### LINKED ISSUE #15356: _shell_utils.py does not return expected result
Under Windows 7, _shell_utils.NativeParser.split (WindowsParser.split) splits executable names containing blanks.

For my Intel Fortran compiler, it should return: ['C:\\Program Files (x86)\\IntelSWTools\\compilers_and_libraries_2018\\windows\\bin\\intel64\\ifort.exe']

Currently, it returns: ['C:\\Program', 'Files', '(x86)\\IntelSWTools\\compilers_and_libraries_2018\\windows\\bin\\intel64\\ifort.exe']

This results in:
```
- info message: "Could not locate executable C:\Program", 
 - incorrect compiler run commands:
compiling Fortran 90 module sources
Fortran f77 compiler: C:\Program Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe /nologo /MD /nbs /names:lowercase /assume:underscore /O1 /arch:IA32 /QaxSSE3
Fortran f90 compiler: C:\Program Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe /nologo /MD /nbs /names:lowercase /assume:underscore /O1 /arch:IA32 /QaxSSE3
Fortran fix compiler: C:\Program Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe Files (x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe /nologo /MD /nbs /names:lowercase /assume:underscore /O1 /arch:IA32 /QaxSSE3
 - warning message after compile instructions (see below)
```
### Error message:
```
ifort.exe:f90: D:\PAJarrige\IstDvp\Gan_API\Python\gandll\pyGanSim.f90
ifort: command line warning #10161: unrecognized source type 'Files'; object file assumed
ifort: command line warning #10161: unrecognized source type '(x86)\IntelSWTools\compilers_and_libraries_2018\windows\bin\intel64\ifort.exe'; object file assumed
```
### Numpy/Python version information:
1.18.0

### PR DESCRIPTION
**Pull request by eric-wieser**
Fixes #15356, I think

### CODE REVIEW

**Comment by pajarrige** [comment]
This fix does not solve the issue in my case - see the attached log.
compiler name should be placed inside square brackets in lines 499, 501, 503.
[build200132patch.log](https://github.com/numpy/numpy/files/4105639/build200132patch.log)

**Comment by eric-wieser** [comment]
Indeed, CI shows the same. I think the first iteration of this that got merged a while ago was not fully thought through (xref https://github.com/numpy/numpy/pull/11898/files#r229861814).

**Comment by eric-wieser** [comment]
I think it's fairly clear this needs a more principled approach that what I was attempting here. I'll come back to this in a few weeks and try and work out what the semantics were before the `exec_command`-removal.

**Comment by charris** [comment]
@eric-wieser Ping.

File: numpy/distutils/fcompiler/__init__.py [hunk 1]
```diff
@@ -472,10 +474,8 @@ def customize(self, dist = None):
         fixflags = []
 
         if f77:
-            f77 = _shell_utils.NativeParser.split(f77)
             f77flags = self.flag_vars.f77
```
Comment (eric-wieser) [inline numpy/distutils/fcompiler/__init__.py]: Some context - the root cause here is that `f77` could either be initialized by:
* `some_legal_f77_command[0]`, in which case it represents a single string
* `os.environ["SOME_ENV_VAR"]`, in which case it represents a string that needs splitting according to shell rules
* some other things probably too

**Comment by mattip** [comment]
I put a 1.22 milestone for this, maybe we can figure out a clean fix by then

**Pull request closed without merging**